businesslogic/__tests__/analytics-profiles.test.ts

```diff
@@ -0,0 +1,228 @@
+import { QueueClient } from "@azure/storage-queue";
+import { TelemetryClient } from "applicationinsights";
+import * as TE from "fp-ts/TaskEither";
+import {
+  INonNegativeIntegerTag,
+  NonNegativeInteger
+} from "@pagopa/ts-commons/lib/numbers";
+import { EmailString, NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import { Producer, ProducerRecord } from "kafkajs";
+
+import { pipe } from "fp-ts/lib/function";
+import * as RA from "fp-ts/ReadonlyArray";
+
+import * as KA from "../../outbound/adapter/kafka-outbound-publisher";
+import * as QAF from "../../outbound/adapter/queue-outbound-mapper-publisher";
+import * as TA from "../../outbound/adapter/tracker-outbound-publisher";
+import * as PDVA from "../../outbound/adapter/pdv-id-outbound-enricher";
+
+import { getAnalyticsProcessorForDocuments } from "../analytics-publish-documents";
+
+import { aFiscalCode } from "../../__mocks__/services.mock";
+import { OutboundPublisher } from "../../outbound/port/outbound-publisher";
+import { sha256 } from "../../utils/pdv";
+import * as pdv from "../../utils/pdv";
+import { RetrievedProfile } from "@pagopa/io-functions-commons/dist/src/models/profile";
+import { PreferredLanguageEnum } from "@pagopa/io-functions-commons/dist/generated/definitions/PreferredLanguage";
+import { ReminderStatusEnum } from "@pagopa/io-functions-commons/dist/generated/definitions/ReminderStatus";
+import { ServicesPreferencesModeEnum } from "@pagopa/io-functions-commons/dist/generated/definitions/ServicesPreferencesMode";
+import { PushNotificationsContentTypeEnum } from "@pagopa/io-functions-commons/dist/generated/definitions/PushNotificationsContentType";
+import { RetrievedProfileWithMaybePdvId } from "../../AnalyticsProfilesChangeFeedInboundProcessorAdapter";
+
+// Data
+
+const aTopic = "a-topic";
+const anError = new Error("An error");
+
+const aCosmosMetadata = {
+  _etag: "_etag",
+  _rid: "_rid",
+  _self: "xyz",
+  _ts: 1
+};
+
+const aRetrievedProfile: RetrievedProfile = {
+  ...aCosmosMetadata,
+  kind: "IRetrievedProfile",
+  id: `${aFiscalCode}-0000000000000002` as NonEmptyString,
+  fiscalCode: aFiscalCode,
+  isEmailEnabled: true,
+  isEmailValidated: true,
+  email: "an-email@email.com" as EmailString,
+  acceptedTosVersion: 22,
+  isInboxEnabled: true,
+  isWebhookEnabled: true,
+  preferredLanguages: [PreferredLanguageEnum.it_IT],
+  pushNotificationsContentType: PushNotificationsContentTypeEnum.FULL,
+  reminderStatus: ReminderStatusEnum.ENABLED,
+  servicePreferencesSettings: {
+    mode: ServicesPreferencesModeEnum.AUTO,
+    version: 1 as NonNegativeInteger
+  },
+  version: 2 as NonNegativeInteger,
+  isTestProfile: false,
+  lastAppVersion: "10.1.1" as RetrievedProfile["lastAppVersion"],
+  _ts: 1637077231000
+};
+const aRetrievedProfileList = [
+  aRetrievedProfile,
+  {
+    ...aRetrievedProfile,
+    id: `${aFiscalCode}-0000000000000003` as NonEmptyString,
+    settingsVersion: 3 as number & INonNegativeIntegerTag,
```

Comment:
typo?
```suggestion
    version: 3 as NonNegativeInteger,
```

Reply:
Fixed in 86cbd75

Reply:
the name of the field needs to be changed too @BurnedMarshal

Reply:
😶‍🌫️ you are right, fixed in f00df91